A51-custom-backend-metrics.md

```diff
@@ -80,170 +70,134 @@ but here are a couple of possible approaches:
+}
 
-// Clear the memory utilization metrics data.
-void deleteMemoryUtilizationMetric();
+function deleteMemoryUtilizationMetric() {
+  // Clear the memory utilization metrics data.
 }
 ```
 
-All the methods above should be thread safe. 
-Allowing updating each metric key individually is convenient for a server application to adopt where it 
-comes with multiple isolated components and each generates their own set of metrics.
+Allowing updating each metric key individually is convenient for a server application to adopt when it 
+has multiple isolated components and each generates their own set of metrics.
 
-A minimum reporting interval (30s by default) could be configured when creating the service.
-The minimum report interval is the lower bound of the OOB metrics report period. It means, in the 
-client OOB request message if the `report_interval` 
-goes below the service configured minimum interval then it is treated as the service minimum interval. 
-If `report_interval` is not specified (equal to 0), it is also treated as service minimum.
+A minimum reporting interval (30s by default) can be configured when creating the service.
+The minimum report interval is the lower bound of the OOB metrics report period. It means, if 
+the `report_interval` from the client OOB request message goes below the service configured minimum 
+interval then it is treated as the service minimum interval. 
+If `report_interval` is not specified (equals to 0), it is also treated as service minimum.
 There is no upper bound, the largest value you can specify is max integer.
-However, it might be useful to have a separate option called default report interval (default 1 minute) 
-that will be used when `report_interval` is not specified (equal to 0), see the code comment above.
-
-Alternatively we can use callback-style APIs in a pull model, but this is suboptimal because 
-it is not performance/scalability friendly.
-In a pull model, the time complexity of doing utilization measurement syscalls reaches O(M*N),
-M being the number of OOB clients and N being their frequency configuration.
-In a push model, the number of syscalls decouples with the number of OOB clients and their configurations.
-In fact, the server application is able to tune the performance by sampling the utilization data at a constant rate.
 
 Internally, the load report server will save each incoming client connection(`responseObserver`) 
-in a `RcaConnection` object or alike, with a custom frequency timer.
-
-```java
-class RcaConnection {
-  StreamObserver<OrcaLoadReport> responseObserver;
-  Timer customFrequencyTimer;
-}
-
-```
-
+in an `RcaConnection` data structure or alike, with a custom frequency timer.
 Initially receiving a new request, or when the response timer fires, the server sends a response to 
 report metrics data. To generate a response, we copy the metrics data saved in the map, 
 but the response proto should only be produced once and be reused.  
```

Comment:
(Actually, I decided to go ahead and implement this optimization in C-core.  But I still don't think we need to call it out in this design, because it is ultimately an implementation detail.)